lib/connectors/memory.js

```diff
@@ -351,26 +351,54 @@ Memory.prototype.fromDb = function(model, data) {
   if (!data) return null;
   data = deserialize(data);
   var props = this._models[model].properties;
+  function cast(prop, val, props) {
+    if (Array.isArray(val)) {
+      return val.map(function(val) {
+        return cast(prop, val, props);
+      });
+    }
+
+    var isArray = Array.isArray(props[prop].type);
```

Comment:
As I mentioned in my comments, `props[prop]` may be undefined when the model has `strict` mode disabled (which is the default).